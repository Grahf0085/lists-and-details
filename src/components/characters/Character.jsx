import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ allies, enemies, photoUrl, name, affiliation }) => (
  <figure>
    <img src={photoUrl} alt={name} />
    <figcaption>
      <p>{name}</p>
      <p>{affiliation}</p>
      <p>{allies}</p>
      <p>{enemies}</p>
    </figcaption>
  </figure>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  allies: PropTypes.array.isRequired,
  enemies: PropTypes.array.isRequired,
  photoUrl: PropTypes.string.isRequired,
  affiliation: PropTypes.string.isRequired,
};

export default Character;
