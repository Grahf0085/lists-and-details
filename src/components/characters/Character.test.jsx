import React from 'react';
import { render } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  it('renders Character', () => {
    const { asFragment } = render(
      <Character
        name="Text Character"
        id="983247fdssf"
        photoUrl="http://image.com/image.png"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
