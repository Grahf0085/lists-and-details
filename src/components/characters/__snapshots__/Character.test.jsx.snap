// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Text Character"
    />
    <figcaption>
      <p>
        Text Character
      </p>
      <p>
        Plasma Nation
      </p>
      <p>
        Gaint Monkey
      </p>
      <p>
        Water Lord
      </p>
    </figcaption>
  </figure>
</DocumentFragment>
`;
